### ISSUE TITLE
Polymer.Collection does not get updated after notifyPath

### ISSUE THREAD

**Issue by tomalec** [Polymer/polymer#2540]
I'm trying to use `.notifyPath` to notify `dom-bind` template after non-polymer data provider changed the data-model. To obtain the `path` to be used I need to use undocumented `Polymer.Collection` that  works...

once.

``` js
// use Polymer's undocumented feature, to make it possible to use public API
var collection = Polymer.Collection.get(model.path_to.array[0].nested_array);
// fetch the key required by .notifyPath method
var old_value = model.path_to.array[0].nested_array[0];
var key_for_initial_value = collection.getKey(old_value);
console.log("Initial value key", key_for_initial_value);
// non-Polymer code changes an object
var first_value = {prop: "firstChange"};
model.path_to.array[0].nested_array[0] = first_value;
// notify dom bind to reflect the change in DOM
domBind.notifyPath("model.path_to.array.0.nested_array." + key_for_initial_value, first_value );
// so far it works, but it worked only for the first time.
```

But when I try to do this once again, `collection.getKey` returns undefined

Here is an example in http://jsbin.com/mezipa/3/edit?html,console,output

**Comment by tomalec** [comment]
The solution is to use (also undocumented) `._notifySplice`:

```js
domBind._notifySplice(model.path_to.array[0].nested_array, "model.path_to.array.0.nested_array", 0, 1, [old_value] );
```
http://jsbin.com/siwuna/edit?html,output

I doesn't look like most intuitive API, especially that public, documented `.notifyPath` seemed to work fine for replacing array properties, and array items for the first time.

I think, even after solving #2509 it would be nice do cleary point such difference in docs.

**Comment by azakus** [comment]
@tomalec Do the docs after #2513 make this any clearer now?

**Comment by tomalec** [comment]
After 1.2 it get even worse.
I change it to use `.` separated path with indexes for `notifyPath` 
and the changes in nested array are not changed at all
http://jsbin.com/gufutu/edit?html,console,output

```js
model.path_to.array[0].nested_array[0] = {prop: "first Change"};
domBind.notifyPath("model.path_to.array.0.nested_array.0", model.path_to.array[0].nested_array[0] );
setTimeout(function(){
    model.path_to.array[0].nested_array[0] = {prop: "second Change"};
    domBind.notifyPath("model.path_to.array.0.nested_array.0", model.path_to.array[0].nested_array[0] );
```

**Comment by tomalec** [comment]
I have created a separate issue for this, as now code docs suggests that I should use `notifySplices` instead of `notifyPath` mentioned in original issue, and hopefully I will not have to bother about `Polymer.Collection` at all.

Just one thing. It's still not 100% clear to me which method should I use to notify about  changing array properties (not items)?

**Comment by TimvdLippe** [comment]
`Polymer.Collection` has been removed and I assume your usecase has been fixed since. If you can still reproduce this issue without `Polymer.Collection` please let us know!
